packages/core/helper-plugin/src/components/DynamicTable.tsx

```diff
@@ -217,69 +237,41 @@ const Table = ({
   );
 };
 
-Table.defaultProps = {
-  action: undefined,
-  children: undefined,
-  components: {
-    ConfirmDialogDeleteAll: undefined,
-    ConfirmDialogDelete: undefined,
-  },
-  footer: undefined,
-  headers: [],
-  isLoading: false,
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  onConfirmDeleteAll() {},
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  onConfirmDelete() {},
-  onOpenDeleteAllModalTrackedEvent: undefined,
-  rows: [],
-  withBulkActions: false,
-  withMainAction: false,
-  renderBulkActionsBar: undefined,
-};
+interface Header {
+  fieldSchema?: {
+    type: string;
+  };
+  name: string;
+  metadatas: {
+    sortable: boolean;
+    label: string;
+    mainField?: {
+      name: string;
+    };
+  };
+}
 
-Table.propTypes = {
-  action: PropTypes.node,
-  children: PropTypes.node,
-  contentType: PropTypes.string.isRequired,
-  components: PropTypes.shape({
-    ConfirmDialogDelete: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
-    ConfirmDialogDeleteAll: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
-  }),
-  footer: PropTypes.node,
-  headers: PropTypes.arrayOf(
-    PropTypes.shape({
-      cellFormatter: PropTypes.func,
-      key: PropTypes.string.isRequired,
-      metadatas: PropTypes.shape({
-        label: PropTypes.string.isRequired,
-        sortable: PropTypes.bool,
-      }).isRequired,
-      name: PropTypes.string.isRequired,
-    })
-  ),
-  isLoading: PropTypes.bool,
-  onConfirmDeleteAll: PropTypes.func,
-  onConfirmDelete: PropTypes.func,
-  onOpenDeleteAllModalTrackedEvent: PropTypes.string,
-  rows: PropTypes.array,
-  withBulkActions: PropTypes.bool,
-  withMainAction: PropTypes.bool,
-  renderBulkActionsBar: PropTypes.func,
-};
+export interface TableHead {
+  areAllEntriesSelected?: boolean;
+  entriesToDelete?: Array<string | number>;
+  headers?: Array<Header>;
+  onSelectAll: BaseCheckbox['onChange'];
+  withMainAction?: boolean;
+  withBulkActions?: boolean;
+}
 
 const TableHead = ({
-  areAllEntriesSelected,
-  entriesToDelete,
-  headers,
+  areAllEntriesSelected = false,
+  entriesToDelete = [],
+  headers = [],
   onSelectAll,
   withMainAction,
   withBulkActions,
-}) => {
+}: TableHead) => {
   const { formatMessage } = useIntl();
   const [{ query }, setQuery] = useQueryParams();
   const sort = query?.sort || '';
-  const [sortBy, sortOrder] = sort.split(':');
+  const [sortBy, sortOrder] = (sort as string).split(':');
```

Comment:
perhaps we can do `const sort = typeof query?.sort === "string" ? query.sort : ''`

So we're certain `sort` is a string instead of telling TS it is :)

Reply:
perhaps we can do `const sort = typeof query?.sort === "string" ? query.sort : ''`

So we're certain `sort` is a string instead of telling TS it is :)